parser/parser.go

```diff
@@ -1162,6 +1183,23 @@ func (p *Parser) parseGo() stmt.Stmt {
 	return g
 }
 
+func (p *Parser) parseVar() *stmt.Var {
+	// a simple 'var f T' or 'var f T = ...'
+	s := &stmt.Var{
+		Position: p.pos(),
+		Name:     p.parseIdent().Name,
+	}
+	if p.s.Token != token.Assign {
+		s.Type = p.parseType()
+	}
+	if p.s.Token == token.Assign {
+		p.expect(token.Assign)
```

Comment:
This line doesn't do anything, because you've checked that p.s.Token == token.Assign already. 

You may want to check for a semicolon at the end, if the caller of parseVar doesn't already. (Makes for nicer error messages if you write something like "var a b c".